activerecord/lib/active_record/autosave_association.rb

```diff
@@ -401,7 +401,10 @@ def save_collection_association(reflection)
                 if autosave
                   saved = association.insert_record(record, false)
                 else
-                  association.insert_record(record) unless reflection.nested?
+                  association_saved = association.insert_record(record) unless reflection.nested?
+                  if reflection.options.fetch(:validate, true)
```

Comment:
Is `validate: nil` considered true?

Reply:
It looks like it is:

https://github.com/rails/rails/blob/df850b8d7126044a820291bbfd2bce80636a442c/activerecord/lib/active_record/autosave_association.rb#L25-L27

We should be calling `reflection.validate?` here though, as `define_autosave_validation_callbacks` does:

https://github.com/rails/rails/blob/df850b8d7126044a820291bbfd2bce80636a442c/activerecord/lib/active_record/autosave_association.rb#L213